python/tchannel/tornado/timeout.py

```diff
@@ -37,7 +37,9 @@ def timeout(future, seconds=1, io_loop=None):
     def raise_timeout(*args, **kwargs):
         if future.running():
             future.set_exception(TimeoutError())
-
-    io_loop.call_later(seconds, raise_timeout)
-
-    yield
+    if seconds == 0:
```

Comment:
What? If there is no timeout, why is this function being called? Zero or negative timeout is not valid input.